### DISCUSSION TITLE
Uploading an app icon with Fastlane

### DISCUSSION THREAD

**Discussion by brianhumphreys** [fastlane/fastlane#17921] [Q&A]
Hello Fastlane community,

I see that the app_icon property in the deliver function is deprecated.  What is the new recommended approach to uploading an app icon using good ol' Fastlane?  Hopefully this is still an available feature!

**Comment by rogerluan** [comment]
Hey @brianhumphreys 👋 

This is now done via Xcode, it's part of your app binary now, under your app's xcassets folder → AppIcon (more specifically the 1024x1024 one)

Hope this helps!

**Reply by brianhumphreys** [reply]
Thanks for the reply @rogerluan,

So does this mean that there is no longer an automated approach to uploading an icon?

**Reply by rogerluan** [reply]
Correct. Here's what Apple says:

>This icon will be used on the App Store. For apps built with Xcode 9 or later, add this icon in the build. For apps built with earlier versions of Xcode, add the icon here [in the App Store Connect website]
https://help.apple.com/app-store-connect/#/dev8b5cb82e2

The "add the icon here" (pre-Xcode 9) step can be done via fastlane, but I would be surprised if that still works, 4 years later 😅
